oak_runtime/src/node/http/user.rs

```diff
@@ -0,0 +1,303 @@
+            &self.privilege,
+        )?;
+
+        // Put the HTTP request message inside the per-invocation request channel.
+        pipe.insert_message(
+            &runtime,
+            outer_http_invocation.request.ok_or(()).map_err(|_err| {
+                warn!("Cannot retrieve request from the OuterHttpInvocation");
+            })?,
+        )?;
+
+        // Send an invocation message (with attached handles) to the Oak Node.
+        trace!(
+            "UserNode {:?}: Sending invocation on outer_http_invocation.invocation_sender",
+            self.node_name
+        );
+        pipe.send_invocation(
+            &runtime,
+            outer_http_invocation
+                .invocation_sender
+                .ok_or(())
+                .map_err(|_err| {
+                    warn!("Cannot retrieve invocation_sender from the OuterHttpInvocation");
+                })?
+                .handle
+                .handle,
+        )?;
+
+        // Close all local handles except for the one that allows reading responses.
+        pipe.close(&runtime);
+
+        Ok(pipe.response_reader)
+    }
+
+    fn try_run(
+        self: Box<Self>,
+        runtime: RuntimeProxy,
+        startup_handle: oak_abi::Handle,
+    ) -> Result<(), ()> {
+        let receiver = Receiver::<OuterHttpInvocation>::new(ReadHandle {
+            handle: startup_handle,
+        });
+        let outer_http_invocation = receiver.receive(&runtime).map_err(|err| {
+            error!(
+                "{}: Failed to retrieve the outer HTTP invocation: {:?}",
+                self.node_name, err
+            )
+        })?;
+
+        // Send the request to the Oak node and get a handle to the channel containing the response.
+        let response_reader =
+            self.inject_http_request(runtime.clone(), outer_http_invocation.clone())?;
+        let response_receiver = crate::io::Receiver::new(ReadHandle {
+            handle: response_reader,
+        });
+        let response = response_receiver.receive(&runtime).map_err(|err| {
+            error!(
+                "Could not receive the response from the Oak node on channel {}: {}.",
+                response_reader, err
+            )
+        })?;
+
+        outer_http_invocation
+            .response_sender
+            .ok_or(())
+            .and_then(|sender| {
+                sender.send(response, &runtime).map_err(|err| {
+                    warn!(
+                        "{}: Could not send response to HTTP server pseudo-node: {}",
+                        self.node_name, err,
+                    )
+                })
+            })
+    }
+}
+
+impl Node for UserNode {
+    fn node_type(&self) -> &'static str {
+        "user-node"
+    }
+
+    fn run(
+        self: Box<Self>,
+        runtime: RuntimeProxy,
+        startup_handle: oak_abi::Handle,
+        _notify_receiver: oneshot::Receiver<()>,
+    ) {
+        let _unit_result = self.try_run(runtime, startup_handle);
```

Comment:
This seems weird that `try_run` returns a result, but is only used in one place where the result is ignored.